securedrop/source.py

```diff
@@ -379,6 +377,15 @@ def login():
             flash("Sorry, that is not a recognized codename.", "error")
     return render_template('login.html')
 
+@app.route('/logout')
+def logout():
+    if logged_in():
+        session.pop('logged_in', None)
```

Comment:
The session also contains the `codename`, which is a sensitive piece of information. Might as well clear the whole thing, all of the data is invalidated once a user logs out. `session.clear()` should do the trick.